gcstorage/gcstorage.go

```diff
@@ -0,0 +1,275 @@
+package gcstorage
+
+import (
+	"cmp"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"github.com/goware/cachestore"
+	"google.golang.org/api/iterator"
+	"google.golang.org/api/option"
+)
+
+const DefaultTTL = time.Second * 24 * 60 * 60 // 1 day in seconds
+
+type cacheObject[T any] struct {
+	Object    T         `json:"object"`
+	ExpiresAt time.Time `json:"expires_at"`
+}
+
+var _ cachestore.Store[any] = &GCStorage[any]{}
+
+type GCStorage[V any] struct {
+	keyPrefix string
+
+	defaultKeyExpiry time.Duration
+
+	client       *storage.Client
+	bucketHandle *storage.BucketHandle
+}
+
+func Backend(cfg *Config, opts ...cachestore.StoreOptions) cachestore.Backend {
+	options := cachestore.ApplyOptions(opts...)
+	cfg.StoreOptions = options
+	return cfg
+}
+
+func NewWithBackend[V any](backend cachestore.Backend, opts ...cachestore.StoreOptions) (cachestore.Store[V], error) {
+	cfg, ok := backend.(*Config)
+	if !ok {
+		return nil, fmt.Errorf("cachestore/gcstorage: invalid backend config supplied")
+	}
+	for _, opt := range opts {
+		opt.Apply(&cfg.StoreOptions)
+	}
+	return New[V](cfg, cfg.StoreOptions)
+}
+
+func New[V any](cfg *Config, opts ...cachestore.StoreOptions) (cachestore.Store[V], error) {
+	for _, opt := range opts {
+		opt.Apply(&cfg.StoreOptions)
+	}
+
+	var gcOpts []option.ClientOption
+	if cfg.Credentials != nil {
+		gcOpts = append(gcOpts, option.WithCredentials(cfg.Credentials))
+	}
+
+	client, err := storage.NewClient(context.Background(), gcOpts...)
+	if err != nil {
+		return nil, fmt.Errorf("cachestore/gcstorage: new client returned error: %w", err)
+	}
+
+	return &GCStorage[V]{
+		keyPrefix:        cfg.KeyPrefix,
+		defaultKeyExpiry: cmp.Or(cfg.DefaultKeyExpiry, DefaultTTL),
```

Comment:
I am getting this warning: `cmp.Or requires go1.22 or later (module is go1.21)`

```suggestion
	if cfg.DefaultKeyExpiry == 0 {
		cfg.DefaultKeyExpiry = DefaultTTL
	}

	return &GCStorage[V]{
		keyPrefix:        cfg.KeyPrefix,
		defaultKeyExpiry: cmp.Or(cfg.DefaultKeyExpiry, DefaultTTL),

```